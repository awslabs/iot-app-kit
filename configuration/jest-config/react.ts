export default {
  setupFilesAfterEnv: ['jest-extended/all'],
  maxWorkers: '50%',
  moduleFileExtensions: ['js', 'ts', 'jsx', 'tsx'],
  collectCoverage: true,
  testEnvironment: 'jsdom',
  transform: {
    '.*\\.(tsx?|jsx?)$': [
      '@swc/jest',
      {
        jsc: {
          transform: {
            react: {
              runtime: 'automatic',
            },
          },
        },
      },
    ],
  },
  transformIgnorePatterns: [],
  moduleNameMapper: {
    '\\.(css|scss)$': 'identity-obj-proxy',
  },
};
