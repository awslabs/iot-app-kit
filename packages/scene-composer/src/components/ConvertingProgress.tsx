import { Box, Container, ProgressBar } from '@cloudscape-design/components';
import React from 'react';
import { useIntl } from 'react-intl';
import styled from 'styled-components';
import * as awsui from '@cloudscape-design/design-tokens';

const ConvertingProgressContainer = styled(Container)`
  width: 100%;
`;

const ConvertingProgressDescriptionBox = styled(Box)`
  color: ${awsui.colorTextBodySecondary}!important;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const ConvertingProgress: React.FC<{ total: number; converted: number }> = ({ total, converted }) => {
  const { formatMessage } = useIntl();

  return (
    <ConvertingProgressContainer data-testid='progress-container'>
      {total > 0 && (
        <ProgressBar
          status='in-progress'
          value={(converted / total) * 100}
          additionalInfo={
            <ConvertingProgressDescriptionBox>
              {formatMessage(
                {
                  defaultMessage: '{convertedNumber} out of {totalNumber} converted',
                  description: 'Progress Bar description',
                },
                { convertedNumber: converted, totalNumber: total },
              )}
            </ConvertingProgressDescriptionBox>
          }
          label={formatMessage({ defaultMessage: 'Converting nodes', description: 'Progress Bar label' })}
        />
      )}
    </ConvertingProgressContainer>
  );
};
