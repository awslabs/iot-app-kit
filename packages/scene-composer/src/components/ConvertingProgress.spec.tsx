import React from 'react';
import { render } from '@testing-library/react';

import { ConvertingProgress } from './ConvertingProgress';

describe('ConvertingProgress', () => {
  it('should render with correct progress with 0% done', () => {
    const { container, queryByText } = render(<ConvertingProgress total={10} converted={0} />);

    const progressBar = queryByText('0 out of 10 converted');
    expect(progressBar).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('should render with correct progress with partially done', () => {
    const { container, queryByText } = render(<ConvertingProgress total={9} converted={2} />);

    const progressBar = queryByText('2 out of 9 converted');
    expect(progressBar).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('should render with correct progress with all done', () => {
    const { container, queryByText } = render(<ConvertingProgress total={10} converted={10} />);

    const progressBar = queryByText('10 out of 10 converted');
    expect(progressBar).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
