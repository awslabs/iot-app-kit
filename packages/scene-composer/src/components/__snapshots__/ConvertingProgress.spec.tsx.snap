// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConvertingProgress should render with correct progress with 0% done 1`] = `
<div>
  <div
    class="sc-aXZVg jRVSTR"
    data-mocked="Container"
    data-testid="progress-container"
  >
    <div
      data-mocked="ProgressBar"
      label="Converting nodes"
      status="in-progress"
      value="0"
    >
      <div>
        <div
          class="sc-gEvEer jzVSqD"
          data-mocked="Box"
        >
          0 out of 10 converted
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConvertingProgress should render with correct progress with all done 1`] = `
<div>
  <div
    class="sc-aXZVg jRVSTR"
    data-mocked="Container"
    data-testid="progress-container"
  >
    <div
      data-mocked="ProgressBar"
      label="Converting nodes"
      status="in-progress"
      value="100"
    >
      <div>
        <div
          class="sc-gEvEer jzVSqD"
          data-mocked="Box"
        >
          10 out of 10 converted
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConvertingProgress should render with correct progress with partially done 1`] = `
<div>
  <div
    class="sc-aXZVg jRVSTR"
    data-mocked="Container"
    data-testid="progress-container"
  >
    <div
      data-mocked="ProgressBar"
      label="Converting nodes"
      status="in-progress"
      value="22.22222222222222"
    >
      <div>
        <div
          class="sc-gEvEer jzVSqD"
          data-mocked="Box"
        >
          2 out of 9 converted
        </div>
      </div>
    </div>
  </div>
</div>
`;
